component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -228,71 +245,120 @@ public String getContextIdentifier(HttpServletRequest httpServletRequest) {
+        configProperties.add(payload);
+
+        Property httpResponse = new Property();
+        httpResponse.setName(SMSOTPConstants.HTTP_RESPONSE);
+        httpResponse.setDisplayName("HTTP Response Code");
+        httpResponse.setRequired(false);
+        httpResponse
+                .setDescription("Enter the HTTP response code the API sends upon successful call. Leave empty if unknown");
+        httpResponse.setDisplayOrder(4);
+        configProperties.add(httpResponse);
 
         return configProperties;
     }
 
-    public boolean sendRESTCall(String url, String urlParameters) throws IOException {
+    public boolean sendRESTCall(String smsUrl, String httpMethod, String headerString, String payload,
+                                String httpResponse) throws IOException, AuthenticationFailedException {
+
         HttpsURLConnection connection = null;
+        String smsMessage = SMSOTPConstants.SMS_MESSAGE;
+
         try {
-            URL smsProviderUrl = new URL(url + urlParameters);
+            smsUrl = smsUrl.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
+                    smsMessage.replaceAll("\\s", "+") + otpToken);
+            URL smsProviderUrl = new URL(smsUrl);
             connection = (HttpsURLConnection) smsProviderUrl.openConnection();
             connection.setDoInput(true);
             connection.setDoOutput(true);
-            connection.setRequestMethod(SMSOTPConstants.HTTP_METHOD);
-            if (connection.getResponseCode() == 200) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Code is successfully sent to your mobile number");
+
+            String[] headerList;
+            if (!headerString.trim().isEmpty()) {
+                headerString = headerString.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg",
+                        smsMessage + otpToken);
+                headerList = headerString.split(",");
+
+                for (int i = 0; i < headerList.length; i++) {
+                    String[] header = headerList[i].split(":");
+                    connection.setRequestProperty(header[0], header[1]);
                 }
-                return true;
             }
-            connection.disconnect();
-        } catch (MalformedURLException e) {
-            if (log.isDebugEnabled()) {
-                log.debug("Invalid URL", e);
+
+            if (httpMethod.trim().toLowerCase().equals(SMSOTPConstants.GET_METHOD.toLowerCase())) {
+                connection.setRequestMethod(SMSOTPConstants.GET_METHOD);
+            } else if (httpMethod.trim().toLowerCase().equals(SMSOTPConstants.POST_METHOD.toLowerCase())) {
+                connection.setRequestMethod(SMSOTPConstants.POST_METHOD);
+                if (!payload.trim().isEmpty()) {
+                    payload = payload.replaceAll("\\$ctx.num", mobile).replaceAll("\\$ctx.msg", smsMessage + otpToken);
+                }
+
+                OutputStreamWriter writer = null;
```

Comment:
Import the class java.io.OutputStreamWriter. This authenticator's build is failing.